src/__tests__/wait-for.js

```diff
@@ -255,3 +255,42 @@ test('the real timers => fake timers error shows the original stack trace when c
 
   expect((await waitForError).stack).not.toMatch(__dirname)
 })
+
+test('allow further async tasks to complete after the MutationObserver callback fired', async () => {
+  jest.useRealTimers()
+  renderIntoDocument(`<div id="async">a</div>`)
+  let waitForCount = 0
+  const el = document.getElementById('async')
+  const update = () => {
+    setTimeout(() => {
+      el.textContent += 'a'
+    }, 1)
+  }
+
+  update()
+  update()
+  update()
+
+  await waitFor(() => {
+    waitForCount++
+    expect(el).toHaveTextContent('aaaa')
+  })
+
+  // initial sync check + mutation check
+  expect(waitForCount).toBe(2)
+})
+
+test('avoid running the next check when the previous check is still pending', async () => {
+  jest.useFakeTimers()
+  let currentlyRunningCount = 0
+
+  await waitFor(async () => {
+    currentlyRunningCount++
+    await new Promise(resolve => {
+      setTimeout(resolve, 1)
```

Comment:
Shouldn't this be greater than `interval` to illustrate that the `callback` always completed before another one is started?

Reply:
The test fails as expected when this line is removed 🤷‍♂️:
https://github.com/testing-library/dom-testing-library/blob/45830f580ced3ae5337af9ace4a9bd65067dfe5c/src/wait-for.js#L140

When using fake timers, we use `setTimeout(r, 0)` in a `while` loop, which is why this test works:
https://github.com/testing-library/dom-testing-library/blob/45830f580ced3ae5337af9ace4a9bd65067dfe5c/src/wait-for.js#L92

I can still increase the timer if you'd prefer.

Reply:
I don't understand how this is related. What are you testing and why is it related to flushing microtasks? My original statement would still apply to real timer usage.

Reply:
I had to add this test to reach 100% testing coverage

https://github.com/testing-library/dom-testing-library/blob/45830f580ced3ae5337af9ace4a9bd65067dfe5c/src/wait-for.js#L140
☝️ this line isn't covered anymore without this test.

I've tweaked the test now, please take another look.